examples/step-77/doc/intro.dox

```diff
@@ -0,0 +1,245 @@
+  SparseDirectUMFPACK if that's what we choose to use to solve a
+  linear system). This operation will generally be called the
+  "setup" operation.
+- A way to solve a linear system $\tilde J_k x = b$ with whatever
+  matrix $\tilde J_k$ was last computed. This operation will generally
+  be called the "solve" operation.
+
+All of these operations need to be provided to KINSOL by
+[std::function](https://en.cppreference.com/w/cpp/utility/functional/function)
+objects that take the appropriate set of arguments and that generally
+return an integer that indicates success (a zero return value) or
+failure (a nonzero return value). Specifically, the objects we will
+access are the
+SUNDIALS::KINSOL::reinit_vector,
+SUNDIALS::KINSOL::residual,
+SUNDIALS::KINSOL::setup_jacobian, and
+SUNDIALS::KINSOL::solve_jacobian_system
+member variables. (See the documentation of these variables for their
+details.) In our implementation, we will use
+[lambda functions](https://en.cppreference.com/w/cpp/language/lambda)
+to implement these "callbacks" that in turn can call member functions;
+KINSOL will then call these callbacks whenever its internal algorithms
+think it is useful.
+
+
+<h3> Details of the implementation </h3>
+
+The majority of the code of this tutorial program is as in step-15,
+and we will not comment on it in much detail. There is really just one
+aspect one has to pay some attention to, namely how to compute $F(U)$
+given a vector $U$ on the one hand, and $J(U)$ given a vector $U$
+separately. At first, this seems trivial: We just take the
+`assemble_system()` function and in the one case throw out all code
+that deals with the matrix and in the other case with the right hand
+side vector. There: Problem solved.
+
+But it isn't quite as simple. That's because the two are not
+independent if we have nonzero Dirichlet boundary values, as we do
+here. The linear system we want to solve contains both interior and
+boundary degrees of freedom, and when eliminating those degrees of
+freedom from those that are truly "free", using for example
+AffineConstraints::distribute_local_to_global(), we need to know the
+matrix when assembling the right hand side vector.
+
+Of course, this completely contravenes the original intent: To *not*
+assemble the matrix if we can get away without it. We solve this
+problem as follows:
+- We set the starting guess for the solution vector, $U_0$, to one
+  where boundary degrees of freedom already have their correct values.
+- This implies that all updates can have zero updates for these
+  degrees of freedom, and we can build both residual vectors $F(U_k)$
+  and Jacobian matrices $J_k$ that corresponds to linear systems whose
+  solutions are zero in these vector components. For this special
+  case, the assembly of matrix and right hand side vectors is
+  independent, and can be broken into separate functions.
+
+There is an assumption here that all linear solves KINSOL will ask for
```

Comment:
"all linear solves KINSOL" sounds odd? "all linear solver in KONSOL"?

Reply:
I think this was grammatically correct -- as in "all of the linear solves that KINSOL asks for are for update-type vectors". But I've rephrased it anyway.